### PR TITLE
feat(Table): refactoring of types, documentation, and accordion file structure

### PR DESCRIPTION
**Pull request by snorrekim**
### Table, TableTr, TableTd
* added `TableDocs` properties
* removed unused `variant` prop
* fixed some prop types
* added typing to `TableContext`

### Accordion
* moved internal accordion files to separate folder `./table-accordion`
* combined `TableAccordionTd` and `TableAccordionTr` files in to `TableAccordionContent` to share identical logic
* renamed `TableAccordionTd` to `TableAccordionContentSingle`
* renamed `TableAccordionTr` to `TableAccordionContentRow` 
* added typing to `TableAccordionContext`

### Other
* `PropertiesTable` now treats values `null` the same as `undefined`

### CODE REVIEW

**Comment by snorrekim** [comment]
The only real change is combining `TableAccordionTr` and `TableAccordionTd` in to one file, and renaming them to `TableAccordionContentRow` and `TableAccordionContentSingle`. Everything else is just more robust typing and moving files around.

**Review by tujoworker — APPROVED** [review]
Nice work ✨ 

I hope its OK for you that I committed a few changes.

File: packages/dnb-eufemia/src/components/table/TableDocs.ts [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+  children: {
+    doc: 'The content of the component.',
+    type: 'React.ReactNode',
+    status: 'required',
+  },
+}
+
+export const TrEventProperties: PropertiesTableProps = {
+  onClick: {
+    doc: 'will emit when user clicks/expands the table row. Returns a native click.  ',
```
Comment (langz) [inline packages/dnb-eufemia/src/components/table/TableDocs.ts:128]:
```suggestion
    doc: 'will emit when user clicks/expands the table row. Returns a native click. ',
```

**Review by langz — APPROVED** [review]
Looks great, 
I think there's a small formatting issue that leads to a bit strange colors, and descriptions, and caps lock in the docs/properties https://eufemia-git-chore-table-refactor-eufemia.vercel.app/uilib/components/table/properties/

I've added a red underline to the parts I'm questioning:



Screenshot 2024-06-14 at 10 51 49

**Comment by snorrekim** [comment]
Yeah, the red color is string. And the "REAQUIRED" text is the most strange one because it has no value, but also need to say that, so right now it's just text.

**Comment by langz** [comment]
> I'll fix the underlined text part.
> 
> And yeah, the red color is string. And the "REQUIRED" text is the most strange one because it has no value, but also need to say that, so right now it's just text, but it's the best suggestion I have for it.

Great, thanks 🙇

**Pull request merged**